### PR TITLE
Manage customized events conflicts

### PR DESCRIPTION
**Pull request by yabirgb**
This PR adds on-demand detection and cleanup for duplicate on-chain events caused by customized history entries. 

The API now distinguishes between two cases: auto-fixable duplicates (same asset/type/subtype/amount/etc., only sequence index differs) and manual-review cases (same asset and direction but the customized event changes data like amount).

The POST endpoint removes only the auto-fixable duplicates (keeping the customized event) and leaves manual-review groups intact, while GET returns both sets so the UI can surface them separately.

Part of https://github.com/rotki/rotki/issues/11334

### CODE REVIEW

File: docs/api.rst [hunk 1]
```diff
@@ -6497,6 +6497,97 @@ Match exchange asset movements with onchain events
...
+
+      HTTP/1.1 200 OK
+      Content-Type: application/json
+
+      {
+          "result": {
+              "auto_fix_group_identifiers": [
+                  "0x7f9b1d9b3d6c80b6f699a25f3e91c408155cbff965d0f3a0df4d3a4f8f4b73c7"
+              ],
+              "manual_review_group_identifiers": [
```
Comment (nicholasyoder) [inline docs/api.rst]:
```suggestion
              "manual_review_group_ids": [
```
These are a bit long, maybe it would be better to shorten it to just `..._ids`? same for the other one.

File: docs/api.rst [hunk 2]
```diff
@@ -6497,6 +6497,97 @@ Match exchange asset movements with onchain events
...
+
+   .. sourcecode:: http
+
+      HTTP/1.1 200 OK
+      Content-Type: application/json
+
+      {
+          "result": {
+              "removed_event_identifiers": [123, 124],
+              "auto_fix_group_identifiers": [],
```
Comment (nicholasyoder) [inline docs/api.rst]: this would always be empty then right? Since this is the response after doing the auto fix. So could be skipped in this response

File: rotkehlchen/tasks/events.py [hunk 3]
```diff
@@ -40,6 +48,53 @@
...
+    notes: str | None
+    event_type: str
+    event_subtype: str
+    extra_data: str | None
+    entry_type: int
+    counterparty: str | None
+    address: str | None
+    customized: bool
+
+    def signature(self) -> tuple:
```
Comment (nicholasyoder) [inline rotkehlchen/tasks/events.py:71]: maybe this should actually hash the values or something? This gets used as a dict key which I expect is slower when its a tuple like this versus a string hash of these values. But maybe any gain from this is negligible.

File: rotkehlchen/tasks/events.py [hunk 4]
```diff
@@ -69,6 +124,151 @@ def process_asset_movements(database: 'DBHandler') -> None:
...
+def find_customized_event_duplicate_groups(
+        database: 'DBHandler',
+        group_identifiers: list[str] | None = None,
+) -> tuple[list[str], list[str], list[int]]:
+    """Return group identifiers for auto-fixable duplicates and manual review groups.
+
+    Auto-fixable groups are exact matches (same fields except sequence_index/identifier)
+    between at least one customized event and at least one non-customized event for EVM/Solana.
+    Manual review groups are same-asset/same-direction pairs between customized and
+    non-customized EVM/Solana events that are not exact matches.
```
Comment (nicholasyoder) [inline rotkehlchen/tasks/events.py:205]: the exact_duplicate_ids seem to not be mentioned here.

File: rotkehlchen/tasks/events.py [hunk 5]
```diff
@@ -69,6 +124,151 @@ def process_asset_movements(database: 'DBHandler') -> None:
...
+                non_customized_ids.add(event.identifier)
+
+        # Collect non-customized duplicates where a customized event shares the same signature.
+        group_exact_ids: set[int] = set()
+        for customized_ids, non_customized_ids in signatures.values():
+            if customized_ids and non_customized_ids:
+                group_exact_ids.update(non_customized_ids)
+
+        # Auto-fixable groups have at least one exact customized/non-customized pair.
+        if group_exact_ids:
```
Comment (nicholasyoder) [inline rotkehlchen/tasks/events.py]:
```suggestion
        if len(group_exact_ids) > 0:
```

File: rotkehlchen/tasks/events.py [hunk 6]
```diff
@@ -69,6 +124,151 @@ def process_asset_movements(database: 'DBHandler') -> None:
...
+            if (direction := event.direction()) is None:
+                continue
+            pair = (event.asset, direction)
+            if event.customized:
+                customized_pairs.add(pair)
+            elif event.identifier not in group_exact_ids:
+                non_customized_pairs.add(pair)
+
+        # Flag groups where customized/non-customized share asset+direction but are not exact.
+        if customized_pairs and non_customized_pairs and customized_pairs & non_customized_pairs:
```
Comment (nicholasyoder) [inline rotkehlchen/tasks/events.py]: this looks confusing. I believe explicitly checking len() on all these would help - I initially was confused about what the & check was doing and thinking `customized_pairs & non_customized_pairs` resolves to a boolean when really it returns another set at least if I am understanding correctly.

File: rotkehlchen/tests/api/test_customized_event_duplicates.py [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+            amount=ONE,
+        )
+        manual_review_customized = EvmSwapEvent(
+            tx_ref=manual_tx_hash,
+            sequence_index=1,
+            timestamp=timestamp,
+            location=Location.ETHEREUM,
+            event_subtype=HistoryEventSubType.SPEND,
+            asset=A_ETH,
+            amount=FVal('2'),
```
Comment (nicholasyoder) [inline rotkehlchen/tests/api/test_customized_event_duplicates.py]: maybe add comment here that its this amount being different that makes it be a manual review event

File: rotkehlchen/tasks/events.py (hunk 3 above)
Comment (yabirgb) [reply rotkehlchen/tasks/events.py:71]: Tuples get also hashed. I believe there is no improvement but we might need to adjust the key depending on performance. Need to play with it a bit

Comment (nicholasyoder) [reply rotkehlchen/tasks/events.py:71]: > Tuples get also hashed

ah true. so is probably fine as is.

File: rotkehlchen/api/v1/schemas.py [hunk 8]
```diff
@@ -4691,3 +4691,7 @@ class UnlinkMatchedAssetMovementSchema(Schema):
 
 class TriggerTaskSchema(AsyncQueryArgumentSchema):
     task = SerializableEnumField(enum_class=TaskName, required=True)
+
+
+class CustomizedEventDuplicatesFixSchema(AsyncQueryArgumentSchema):
+    group_identifiers = fields.List(fields.String(), load_default=None)
```
Comment (prettyirrelevant) [inline rotkehlchen/api/v1/schemas.py]:
```suggestion
    group_identifiers = fields.List(NonEmptyStringField(), load_default=None)
```

File: rotkehlchen/api/rest.py [hunk 9]
```diff
@@ -3539,6 +3540,54 @@ def get_unmatched_asset_movements(self, only_ignored: bool) -> Response:
...
+                'removed_event_identifiers': [],
+                'auto_fix_group_ids': auto_fix_group_ids,
+                'manual_review_group_ids': manual_review_group_ids,
+            })
+
+        error = events_db.delete_history_events_by_identifier(
+            identifiers=duplicate_ids,
+            force_delete=False,
+        )
+        if error is not None:
```
Comment (prettyirrelevant) [inline rotkehlchen/api/rest.py]:
```suggestion
        if (error := events_db.delete_history_events_by_identifier(
            identifiers=duplicate_ids,
            force_delete=False,
        )) is not None:
```

File: rotkehlchen/tasks/events.py [hunk 10]
```diff
@@ -69,6 +124,153 @@ def process_asset_movements(database: 'DBHandler') -> None:
...
+                non_customized_ids.add(event.identifier)
+
+        # Collect non-customized duplicates where a customized event shares the same signature.
+        group_exact_ids: set[int] = set()
+        for customized_ids, non_customized_ids in signatures.values():
+            if customized_ids and non_customized_ids:
+                group_exact_ids.update(non_customized_ids)
+
+        # Auto-fixable groups have at least one exact customized/non-customized pair.
+        if len(group_exact_ids) > 0:
```
Comment (prettyirrelevant) [inline rotkehlchen/tasks/events.py:246]: a single `group_id` can appear in both sets if the group has multiple event pairs, some being exact duplicates (auto fix) and others only matching on asset+direction (manual review).

is that expected?

File: rotkehlchen/tests/api/test_customized_event_duplicates.py
Comment (prettyirrelevant) [inline rotkehlchen/tests/api/test_customized_event_duplicates.py:1]: this is missing a test for checking when `group_identifiers` is None and it does an autofix

File: rotkehlchen/tasks/events.py (hunk 10 above)
Comment (nicholasyoder) [reply rotkehlchen/tasks/events.py:246]: mmm... I believe this should be fine unless I'm missing something. Doing an auto fix on such a group would simply fix the ones that can be auto fixed and the ones needing manual review would remain unchanged. We can discuss in discord though if there's something I'm not considering that would make a problem.

File: rotkehlchen/tests/api/test_customized_event_duplicates.py
Comment (nicholasyoder) [reply rotkehlchen/tests/api/test_customized_event_duplicates.py:1]: adjusted the test to also check this.

File: rotkehlchen/tasks/events.py (hunk 10 above)
Comment (nicholasyoder) [reply rotkehlchen/tasks/events.py:246]: As discussed in [discord](https://discord.com/channels/657906918408585217/1460317739540418672/1462853587951554786), changed this to not include such group ids in the auto fix list.
